### PR TITLE
:robot: test sysextension on uki

### PR DESCRIPTION
**Pull request by Itxaka**
Signed-off-by: Itxaka <itxaka@kairos.io>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: tests/assets/sysext/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+work.raw contains a simple script called `hello.sh` that prints "Hello World" to the console.
+hello-broke.raw contains a simple script called `hello.sh` that prints "Hello World" to the console, but it is NOT signed or verity checked.
+
+Both extensions need to have a `extension-release.NAME` with `ID=_any` for it to be identified as a sysextension.
+Full path on the image should be `usr/lib/extension-release.d/extension-release.NAME`
+
+work.raw is verity+signed with the db.key and db.crt test keys found under tests/assets/keys
+
+The idea is to copy them intot he Kairos iso overlay folder and test the verity+signed sysextension loading.
```
Comment (jimmykarily) [inline tests/assets/sysext/README.md:11]:
```suggestion
The idea is to copy them into the Kairos iso overlay folder and test the verity+signed sysextension loading.
```

File: tests/assets/sysext/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+immucore should only copy the valid ones and ignore the invalid ones. A warning should be logged int he immucore log.
```
Comment (jimmykarily) [inline tests/assets/sysext/README.md:13]:
```suggestion
immucore should only copy the valid ones and ignore the invalid ones. A warning should be logged in the immucore log.
```

File: tests/assets/sysext/README.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+This folder contains 2 sysextensions
+
+work.raw contains a simple script called `hello.sh` that prints "Hello World" to the console.
+hello-broke.raw contains a simple script called `hello.sh` that prints "Hello World" to the console, but it is NOT signed or verity checked.
```
Comment (jimmykarily) [inline tests/assets/sysext/README.md:4]: Maybe it's a good idea to have the "bad" extension print a different text, to be sure we got the output for the "good" extensions. Maybe "Hello evil world" ?

File: tests/assets/sysext/README.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+
+
+The sysextensions are really stupid, its just a /usr/local/bin/ dir with a hello.sh script on them.
+work.raw was built with systemd-repart so it would be verity+signed
+```bash
+systemd-repart -S -s SOURCE_DIR OUTPUT_FILE --private-key=tests/assets/keys/db.key --certificate=tests/assets/keys/db.pem
+```
+
+The other one was built with [sysext-bakery](https://github.com/flatcar/sysext-bakery) which makes it wasy to build sysextensions, but doesnt have support for signing or verity yet. So its simple to generate images with it but they wont work on UKI.
```
Comment (jimmykarily) [inline tests/assets/sysext/README.md:32]:
```suggestion
The other one was built with [sysext-bakery](https://github.com/flatcar/sysext-bakery) which makes it easy to build sysextensions, but doesn't have support for signing or verity yet. So its simple to generate images with it but they wont work on UKI.
```

File: tests/uki_test.go [hunk 5]
```diff
@@ -168,12 +169,66 @@ var _ = Describe("kairos UKI test", Label("uki"), Ordered, func() {
...
 		vm.EventuallyConnects(1200)
 
+		By("Checking the boot mode (recovery)", func() {
+			out, err := vm.Sudo("stat /run/cos/recovery_mode")
+			Expect(err).ToNot(HaveOccurred(), out)
+		})
+
+		By("Checking sysext was not copied during boot", func() {
+			out, err := vm.Sudo("stat /.extra/sysext")
+			Expect(err).To(HaveOccurred(), out)
```
Comment (jimmykarily) [inline tests/uki_test.go:229]: Or match the error explicitly (to make sure we got the expected one): https://github.com/onsi/gomega/blob/0e69083804979327b6008a5c7aaab402cd589c5a/matchers/match_error_matcher_test.go#L71

**Review by jimmykarily — APPROVED** [review]

File: tests/assets/sysext/README.md (hunk 3 above)
Comment (Itxaka) [reply tests/assets/sysext/README.md:4]: Checks are done in a different way to not have to relay on the output of the script. Like check if it was copied, check if it was loaded and such.

**Pull request merged**